fooof/objs/fit.py

```diff
@@ -774,14 +779,18 @@ def _robust_ap_fit(self, freqs, power_spectrum):
         freqs_ignore = freqs[perc_mask]
         spectrum_ignore = power_spectrum[perc_mask]
 
+        # Get bounds for aperiodic fitting, dropping knee bound if not set to fit knee
+        ap_bounds = self._ap_bounds if self.aperiodic_mode == 'knee' \
+            else tuple(bound[0::2] for bound in self._ap_bounds)
```

Comment:
Could this check be done in `__init__` since that is where `aperiodic_mode` is set from? That way this code doesn't need to be in two places.

Reply:
The reason I didn't put it it init is that the aperiodic_mode at init time might not be available, or correct. For example, data could be loaded in a FOOOF object, with null settings, and then `add_settings` is used to update the settings for re-fitting. I don't love the duplicated code, but I don't think this check is best in init, so this seems fine to me.